tempest/api/workloadmgr/cli/test_wlm_cli.py

```diff
@@ -276,3 +276,221 @@ def test_05_create_workload_setting_with_invalid_values(self):
+                reporting.add_test_step(
+                    "Verify workload setting name after updation", tvaultconf.PASS)
+            else:
+                reporting.add_test_step(
+                    "Verify workload setting name after updation", tvaultconf.FAIL)
+            if (cli_parser.cli_response_parser(out, 'value') == update_val):
+                reporting.add_test_step(
+                    "Verify workload setting value after updation", tvaultconf.PASS)
+            else:
+                reporting.add_test_step(
+                    "Verify workload setting value after updation", tvaultconf.FAIL)
+            if (cli_parser.cli_response_parser(out, 'hidden') == 'False'):
+                reporting.add_test_step(
+                    "Verify workload setting get-hidden after updation", tvaultconf.PASS)
+            else:
+                reporting.add_test_step(
+                    "Verify workload setting get-hidden after updation", tvaultconf.FAIL)
+
+        except Exception as e:
+            LOG.error("Exception: " + str(e))
+            reporting.set_test_script_status(tvaultconf.FAIL)
+        finally:
+            reporting.test_case_to_write()
+
+    @decorators.attr(type='workloadmgr_cli')
+    def test_07_update_workload_setting_with_invalid_values(self):
+        reporting.add_test_script(str(__name__) + "_update_workload_setting_cli_with_invalid_values")
+        try:
+            cli_error_string = tvaultconf.wl_setting_update_cli_error_string
+            wl_setting_update = command_argument_string.workload_setting_update + " " + workload_setting_name + \
+                                " value45 --description --type set2"
+            LOG.debug("Workload setting update command with invalid values: {}".format(wl_setting_update))
+            rc = cli_parser.cli_returncode(wl_setting_update)
+            if rc == 0:
+                reporting.add_test_step(
+                    "Execute workload setting-update with invalid values", tvaultconf.FAIL)
+                raise Exception("workload setting-update command did not execute correctly")
+            else:
+                reporting.add_test_step(
+                    "Execute workload setting-update with invalid values", tvaultconf.PASS)
+                LOG.debug("workload setting-update command executed correctly")
+
+            cli_err = cli_parser.cli_error(wl_setting_update)
+            LOG.debug("cli error: {}".format(cli_err))
+            if (cli_err and cli_error_string in cli_err):
+                reporting.add_test_step(
+                    "Verify correct error thrown for invalid parameter", tvaultconf.PASS)
+                LOG.debug("Command executed correctly")
+            else:
+                reporting.add_test_step(
+                    "Verify correct error thrown for invalid parameter", tvaultconf.FAIL)
+                reporting.set_test_script_status(tvaultconf.FAIL)
```

Comment:
In failure case instead of raising an exception, you wrote set_test_script_status() Is this because you wanted to execute the code for cleanup (delete workload setting) ?
Please confirm.

Reply:
That is correct.